### PR TITLE
Updated headers and navbar

### PR DESCRIPTION
**Pull request by fernandonevarez**
I restructured the fundamentals of the filing systems to a new design to allow custom headers for each page. However, I also updated the navbar to look like what we had on the old website. Additionally, I made the navbar and headers responsive for both mobile and desktop.

### CODE REVIEW

File: src/components/headers/_headerLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from "react";
```
Comment (reharri7) [inline src/components/headers/_headerLayout.tsx:1]: These are probably fine, but ESLint may prefer ' over ". If it does complain about it then they will have to change

File: src/components/headers/_headerLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+// headers
+import HomePageHeader from "./home";
+import AboutPageHeader from "./about";
+import ContactPageHeader from "./contact";
+
+// wave divider
+import Divider from "../page_layout/pageDivider";
+
+const PageHedaer = () => {
```
Comment (reharri7) [inline src/components/headers/_headerLayout.tsx]: Misspelling

File: src/components/headers/contact.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+
+import headerStyles from "./../../styles/components/headers/_headerLayout.module.scss";
+
+const ContactPageHeader = () => {
+  return (
+    <div id={headerStyles.header_content}>
+      <h1 id={headerStyles.title}>Contact Us</h1>
+      <p id={headerStyles.description}>
+        Get in touch and let us know how we can help you with your cooking needs. Fill out the form and we&#180;ll be in touch as soon as possible.
```
Comment (reharri7) [inline src/components/headers/contact.tsx]: Cooking => coding?

File: src/components/page_layout/_pageLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        />
+        <meta name="author" content="" />{" "}
+        {/* TODO: add all the developers that worked on the website at the end */}
+        <meta name="viewport" content="width=device-width,initial-scale=1" />
+        <link rel="icon" href="/favicon.png" />{" "}
+        {/* TODO: Add website favicon */}
+      </Head>
+
+      {navbar && <Navbar />}
+      <PageHedaer />
```
Comment (reharri7) [inline src/components/page_layout/_pageLayout.tsx]: Misspelling

File: src/components/page_layout/navbars/_navbarLayout.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import DesktopNavbar from "./desktop";
+import MobileNavbar from "./mobile";
+
+const Navbar = () => {
+  const [isMobile, setIsMobile] = useState(false);
+
+  useEffect(() => {
+    // Check if the user is on a mobile device
+    const checkIfMobile = () => {
+      if (typeof window !== "undefined" && window.innerWidth < 768) {
```
Comment (reharri7) [inline src/components/page_layout/navbars/_navbarLayout.tsx:13]: Use 
```suggestion
      SetIsMobile(resultOfThisBoolean)
```

File: src/pages/_app.tsx [hunk 6]
```diff
@@ -15,7 +16,7 @@ const MyApp: AppType<{ session: Session | null }> = ({
   return (
     <SessionProvider session={session}>
       <Layout>
-        {/* This becomes the chidlren of the layout component because it's nested within layout */}
+        {/* This becomes the chidlren of the layout component because it's nested within the layout component*/}
```
Comment (reharri7) [inline src/pages/_app.tsx]: Misspelling

File: src/styles/components/page_layout/navbars/mobile.module.scss [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        justify-content: center;
+        background: $asu_maroon;
+        padding: 1rem 0;
+        .item_wrapper {
+          display: flex;
+          flex-direction: row;
+          flex-wrap: nowrap;
+          align-items: center;
+          width: 60%;
+          // styles that will be appied to bothe the icon and the text
```
Comment (reharri7) [inline src/styles/components/page_layout/navbars/mobile.module.scss]: Misspelling

File: src/styles/components/page_layout/navbars/mobile.module.scss [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    transform: translateX(-100%);
+    opacity: 0.9;
+  }
+
+  // close styles
+  #close {
+    transition: all 700ms ease-in-out;
+    position: absolute;
+    top: 0;
+    // left: 0;
```
Comment (reharri7) [inline src/styles/components/page_layout/navbars/mobile.module.scss]: Remove unused style

File: src/styles/components/page_layout/navbars/mobile.module.scss [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    width: 100vw;
+    height: 100vh;
+    // background: $asu_grey;
```
Comment (reharri7) [inline src/styles/components/page_layout/navbars/mobile.module.scss]: Removed unused style

File: src/styles/components/page_layout/navbars/mobile.module.scss [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+    // opacity: 0.3;
+  }
```
Comment (reharri7) [inline src/styles/components/page_layout/navbars/mobile.module.scss:142]: Remove unused style

File: src/styles/components/page_layout/pageDivider.module.scss [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+//   width: calc(161% + 1.3px);
+//   height: 235px;
+// }
+
+// .custom-shape-divider-bottom-1671265681 .shape-fill {
+//   fill: #ffffff;
+// }
+
+@use "./../../util/vars.scss" as *;
+// another svg styles
```
Comment (reharri7) [inline src/styles/components/page_layout/pageDivider.module.scss]: Remove all above unused styles

Comment (reharri7) [reply src/styles/components/page_layout/pageDivider.module.scss]: You can leave a commented out note with which commit number they were removed in

**Review by reharri7 — CHANGES_REQUESTED** [review]
Overall this is good. I haven't used react in over 2 years so I'm a little foggy on the state management stuff and I'm not up to date with how things work in react nowadays. Check with ESLint and make sure those rules are being enforced. Fix the minor issues in the code and will be able to merge

**Comment by fernandonevarez** [comment]
Alright, all the changes are done except for one 
- ` if (typeof window !== "undefined" && window.innerWidth < 768) {
      SetIsMobile(resultOfThisBoolean) }`

Can't seem to understand the difference between what you're asking for me to change to what I have already.

File: src/components/headers/_headerLayout.tsx (hunk 1 above)
Comment (fernandonevarez) [reply src/components/headers/_headerLayout.tsx:1]: EsLint has no problem with it.

**Pull request merged**